Code/Framework/AzTest/AzTest/AzTest.h

```diff
@@ -448,18 +448,31 @@ int main(int argc, char** argv)
 
 #define DEFAULT_UNIT_TEST_ENV AZ::Test::DefaultTestEnv()
 
-#define AZ_UNIT_TEST_HOOK_1(_1) \
-    AZ_UNIT_TEST_HOOK_ENV(_1) \
-    AZ_BENCHMARK_HOOK() \
+#define AZ_UNIT_TEST_HOOK_1(_1)                                                                                  \
```

Comment:
Is all this actually part of your changes?

Reply:
Yes, this is specifically to deal with an OpenSSL quirk in which it will prevent the module that loaded it from unloading. This caused some very unpleasant teardown issues that required this machinery to deal with.

Reply:
Once Openssl `SSL_CTX_new` is invoked, it prevents the library that made that call from unloading.
Here is the description for that issue: https://github.com/openssl/openssl/issues/3777

The problem is that `AzNetworking.Tests.dll` would load `AzNetworking.dll` because it is statically linked.
The `AZ::Environment` instance being attached to the `AzNetworking.dll` was created within the `AzNetworking.Test.dll` memory space.

Now when the AzTestRunner unloads the `AzNetworking.Tests.dll`, that `AzNetworking.Tests.dll` should unload the `AzNetworking.dll`, but because Openssl is preventing the `AzNetworking.dll` from unloading, only the `AzNetworking.Tests.dll` unloads.

The problem with this is that because `AzNetworking.dll` didn't unload, it still has the `AZ::Environment` that was created within the **AzNetworking.Tests.dll** attached to it, but the `AzNetworking.Tests.dll` has unloaded.
So the the `AzNetworking.dll` actually unloads at the `AzTestRunner` executable shutdown it crashes when trying to destruct static AZ::EnvironmentVariables

The avoid this issue with the `AzNetworking.dll` hanging on to an `AZ::Environment` instance of a unloaded dll, instead we have the AzTestRunner create the canonical `AZ::Environment` instance and then attach the loaded Test module with that instance.
In this case the `AzTestRunner.exe` attaches `AZ::Environment` created within it to the `AzNetworking.Tests.dll`, which subsequently attaches that `AZ::Environment` to the `AzNetowrking.dll`.

Because now the `AzNetworking.dll` has the `AzTestRunner.exe` AZ::Environment instance attached to it, that instance will be alive when the `AzNetworking.dll` unloads when the `AzTestRunner.exe` unloads.

Now with @burelc-amzn [Allocator Availability](https://github.com/o3de/sig-core/pull/31/files#diff-b4326f61ea81bd70570f1f27143f9bfe6d275a93a905002e11426bb5b80314e6) work, the AZ::Environment will be moved to the O3DE Kernel shared library that would be static linked directly in the executable that is running, avoiding the need to attach the AZ::Environment and making it always available.

That would solve this problem once and for all